pennylane/_qubit_device.py

```diff
@@ -1002,7 +1085,9 @@ def sample(self, observable, shot_range=None, bin_size=None):
         if bin_size is None:
             return samples
 
-        return samples.reshape((bin_size, -1))
+        # Split up the last axis into bin-sized and number of bins
+        new_shape = qml.math.shape(samples)[:-1] + (bin_size, -1)
+        return samples.reshape(new_shape)  # TODO: Check with broadcasting
```

Comment:
This branch actually remains unexplored with broadcasting as far as I am aware, because we do not support shot vectors.